### PR TITLE
Eliminate extraneous parens in regex

### PR DESCRIPTION
**Pull request by jim-krueger**

### CODE REVIEW

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#11893

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java [hunk 1]
```diff
@@ -85,9 +85,18 @@ public URITemplate(String theTemplate, List<Parameter> params) {
...
                     customVariables.add(var.getName());
-                    patternBuilder.append('(');
-                    patternBuilder.append(var.getPattern());
-                    patternBuilder.append(')');
+                    String pattern = var.getPattern();
+                    // #11893 Add parenthesis to the pattern to identify a regex in the pattern, 
+                    // however do not add them if they already exist since that will cause the Matcher
+                    // to create extraneous values.  Parens identify a group so multiple parens would
+                    // indicate multiple groups.
+                    if ((pattern != null) && ((pattern.startsWith("(")) && (pattern.endsWith(")")))) {
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java:93]: The null check is redundant.  We already checked it above

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java [hunk 2]
```diff
@@ -85,9 +85,18 @@ public URITemplate(String theTemplate, List<Parameter> params) {
                 variables.add(var.getName());
                 if (var.getPattern() != null) {
                     customVariables.add(var.getName());
-                    patternBuilder.append('(');
-                    patternBuilder.append(var.getPattern());
-                    patternBuilder.append(')');
+                    String pattern = var.getPattern();
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java:88]: // Liberty change begin and // Liberty change end comments are missing

File: dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java (hunk 1 above)
Comment (jhanders34) [reply dev/com.ibm.ws.org.apache.cxf.cxf.rt.frontend.jaxrs.3.2/src/org/apache/cxf/jaxrs/model/URITemplate.java:93]: Also definitely overuse of parentheses in the if condition.

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

**Pull request merged**